test/k8sT/Policies.go

```diff
@@ -719,6 +726,119 @@ var _ = Describe("K8sPolicyTest", func() {
+				Expect(err).To(BeNil())
+				for _, ep := range endpoints {
+					if ep.Status.Networking.Addressing[0].IPV4 == app1PodIP {
+						break
+					}
+				}
+			})
+
+			BeforeEach(func() {
+
+			})
+			AfterEach(func() {
+				kubectl.Exec(fmt.Sprintf("%s annotate pod %s -n %s %s-", helpers.KubectlCmd, appPods[helpers.App1], namespaceForTest, annotation.ProxyVisibility))
+			})
+
+			checkProxyRedirection := func(app1PodIP string, redirected bool) {
+				var not = " "
+
+				if !redirected {
+					not = " not "
+				}
+
+				monitorFile := fmt.Sprintf(monitorFileName, uuid.NewUUID().String())
+
+				By("Starting monitor and generating traffic which should%s redirect to proxy", not)
+				monitorStop := kubectl.MonitorStart(helpers.KubeSystemNamespace, ciliumPod, monitorFile)
+
+				// Let the monitor get started since it is started in the background.
+				time.Sleep(2 * time.Second)
+				curlCmd := helpers.CurlFail(fmt.Sprintf("http://%s/public", app1PodIP))
+				res := kubectl.ExecPodCmd(
+					namespaceForTest, appPods[helpers.App2],
+					curlCmd)
+				// Give time for the monitor to be notified of the proxy flow.
+				time.Sleep(2 * time.Second)
+				monitorStop()
+				ExpectWithOffset(1, res.WasSuccessful()).To(BeTrue(), "%q cannot curl %q", appPods[helpers.App2], app1PodIP)
+				monitorPath := fmt.Sprintf("%s/%s", helpers.ReportDirectoryPath(), monitorFile)
+				By("Reading the monitor log at %s", monitorPath)
+				monitorOutput, err := ioutil.ReadFile(monitorPath)
+				ExpectWithOffset(1, err).To(BeNil(), "Could not read monitor log")
+
+				By("Checking that aforementioned traffic was%sredirected to the proxy", not)
+				reStr := fmt.Sprintf("verdict Forwarded GET http://%s/public", app1PodIP)
+				re := regexp.MustCompile(reStr)
+				out := re.Find(monitorOutput)
+				if redirected {
+					ExpectWithOffset(1, out).ToNot(BeNil(), "traffic was not redirected to the proxy when it should have been")
+				} else {
+					ExpectWithOffset(1, out).To(BeNil(), "traffic was redirected to the proxy when it should have not been redirected")
+				}
+			}
+
+			It("Tests proxy visibility without policy", func() {
+
+				checkProxyRedirection(app1PodIP, false)
+
+				By("Annotating %s with <Ingress/80/TCP/HTTP>", app1Pod)
+				res := kubectl.Exec(fmt.Sprintf("%s annotate pod %s -n %s %s=\"<Ingress/80/TCP/HTTP>\"", helpers.KubectlCmd, app1Pod, namespaceForTest, annotation.ProxyVisibility))
+				res.ExpectSuccess("annotating pod with proxy visibility annotation failed")
+				Expect(kubectl.CiliumEndpointWaitReady()).To(BeNil())
+
+				checkProxyRedirection(app1PodIP, true)
+
+				By("Removing proxy visibility annotation on %s", app1Pod)
+				kubectl.Exec(fmt.Sprintf("%s annotate pod %s -n %s %s-", helpers.KubectlCmd, app1Pod, namespaceForTest, annotation.ProxyVisibility)).ExpectSuccess()
+				Expect(kubectl.CiliumEndpointWaitReady()).To(BeNil())
+
+				checkProxyRedirection(app1PodIP, false)
+
```

Comment:
I recall you mentioned that you've tested the following:

1. No enforcement policy
1. Apply visibility policy
1. Apply enforcement policy that drops the traffic affected by visibility
1. Now the redirect no longer occurs

Are you planning to add this to the regression suite as well?

Reply:
Added this test.